polars/polars-core/src/chunked_array/kernels/strings.rs

```diff
@@ -1,7 +1,17 @@
 use crate::prelude::Arc;
-use arrow::array::{ArrayRef, LargeStringArray, UInt32Array};
+use arrow::array::{ArrayRef, UInt32Array, Utf8Array};
+use arrow::buffer::Buffer;
+use arrow::datatypes::DataType;
 
-pub(crate) fn string_lengths(array: &LargeStringArray) -> ArrayRef {
-    let array: UInt32Array = array.iter().map(|v| v.map(|v| v.len() as u32)).collect();
+pub(crate) fn string_lengths(array: &Utf8Array<i64>) -> ArrayRef {
+    let values = array
+        .offsets()
+        .iter()
+        .windows(2)
+        .map(|x| (x[1] - x[0]) as u32);
+
+    let values = Buffer::from_trusted_len_iter(values);
+
+    UInt32Array::from_data(DataType::UInt32, values, array.validity().clone());
```

Comment:
Just curious. Why does this specify the dtype twice? As in the the type of the array: `UInt32Array` and in the `from_data` call. Is that because of the distinction between logical and physical types?

Reply:
exactly. `UIntArray32` is a type def for `PrimitiveArray<u32>`, and a struct,  `DataType::UInt32` is a variant of the enum for the logical type. Here they are the same; the case is more relevant in e.g.

```
let data_type =  DataType::Timestamp(TimeUnit::Nanosecond, Some("tz"));
Int64Array::from_data(data_type, values, array.validity().clone());
```

`from_data` panics if the datatype is not supported by the physical type (these types are part of the spec and cause UB to consumers of the arrays if passed incorrectly, both on FFI and IPC).